tests/ui/type/pattern_types/signed_ranges.stderr

```diff
@@ -0,0 +1,49 @@
+error[E0277]: the trait bound `u32: Neg` is not satisfied
+  --> $DIR/signed_ranges.rs:6:34
+   |
+LL | type Sign = pattern_type!(u32 is -10..);
+   |                                  ^^^ the trait `Neg` is not implemented for `u32`
+   |
+   = help: the following other types implement trait `Neg`:
+             &f128
+             &f16
+             &f32
+             &f64
+             &i128
+             &i16
+             &i32
+             &i64
+           and 12 others
+
+error[E0277]: the trait bound `char: Neg` is not satisfied
+  --> $DIR/signed_ranges.rs:9:41
+   |
+LL | type SignedChar = pattern_type!(char is -'A'..);
+   |                                         ^^^^ the trait `Neg` is not implemented for `char`
+
+error[E0277]: the trait bound `u8: Neg` is not satisfied
+  --> $DIR/signed_ranges.rs:14:9
+   |
+LL |         -10..253 => {}
+   |         ^^^ the trait `Neg` is not implemented for `u8`
```

Comment:
I tried improving this error by having it point to where we got the `u8` type information from (the scrutinee), but adding the right obligations for that didn't work out, so I'll need to do some more debugging.

But this is also the exact same diagnostic we get on stable and beta, so...